thunder/tests/test_grad.py

```diff
@@ -1183,9 +1183,26 @@ def snippet_phantom_grad_vs_torch_consistency(op, torch_op, sample, comp, singul
 
     args, kwargs = sample.args, sample.kwargs
 
+    def is_output_differentiable(x):
+        # These are the differentiable dtypes in PyTorch.
+        return x.dtype.is_floating_point or x.dtype.is_complex
```

Comment:
Is that really sufficient? Shall we test binary ops with only one input requiring grad?

Reply:
I don't think that will be a problem. Even in that case, we can get the grad for one of the inputs. Did you have something else in mind?

```python
>>> import torch
>>> x = torch.rand(3, 3, requires_grad=True)
>>> y = torch.rand(3, 3, requires_grad=False)
>>> o = x + y
>>> o
tensor([[1.8362, 0.3462, 0.8109],
        [0.1791, 1.6089, 1.4538],
        [1.2674, 1.1876, 1.6030]], grad_fn=<AddBackward0>)
```

Reply:
Also, we filter the inputs which require_grad before passing it to `torch.autograd.grad` 

Ref: https://github.com/Lightning-AI/lightning-thunder/blob/843b2c265ec335aa1406f3d5650324567b88bf78/thunder/tests/test_grad.py#L1203-L1204

Reply:
The filtering takes place in the tests. What happens if I define an op with two floating point inputs and `put_grad` for only one of them?

Reply:
I think we can also check `grad_fn is not None` on the output. Will update to this.

```python
>>> import torch
>>> x = torch.ones(3, requires_grad=True)
>>> x + x
tensor([2., 2., 2.], grad_fn=<AddBackward0>)
>>> (x + x).grad_fn
<AddBackward0 object at 0x7f0502edcf40>
>>> y = torch.ones(3, requires_grad=False)
>>> (y + y).grad_fn
>>> (y + x).grad_fn
<AddBackward0 object at 0x7f0502e21060>
>>> x = torch.ones(3, requires_grad=True)
>>> x < 1
tensor([False, False, False])
>>> (x < 1).grad_fn
>>> torch.topk(x, k=2)
torch.return_types.topk(
values=tensor([1., 1.], grad_fn=<TopkBackward0>),
indices=tensor([0, 1]))
>>> torch.topk(torch.ones(3, requires_grad=False), k=2)
torch.return_types.topk(
values=tensor([1., 1.]),
indices=tensor([0, 1]))
```

Reply:
> The filtering takes place in the tests. What happens if I define an op with two floating point inputs and put_grad for only one of them?

In that case, we compute `grad` for only the outputs which are differentiable. In case of `topk`, we will pass the gradients of input w.r.t. `values` and not the indices.

Reply:
Ping @nikitaved

Reply:
Gentle ping @nikitaved